src/snowflake/cli/plugins/nativeapp/codegen/sandbox.py

```diff
@@ -0,0 +1,201 @@
+    SYSTEM_PATH = 3  # search for a python interpreter in the system path
+    CURRENT = 4  # Use the python interpreter that is currently executing (i.e. `sys.executable`)
+
+
+def execute_script_in_sandbox(
+    script_source: str,
+    env_type: ExecutionEnvironmentType = ExecutionEnvironmentType.AUTO_DETECT,
+    cwd: Optional[Union[str, Path]] = None,
+    timeout: Optional[int] = None,
+    **kwargs,
+) -> subprocess.CompletedProcess:
+    """
+    Executes a python script in a sandboxed environment, and returns its output. The script is executed in a different
+    process. The execution environment is determined by the `env_type` argument. By default, the logic will attempt
+    to auto-detect the correct environment by looking for an active venv or conda environment. If none can be found, it
+    will use the system's default python executable, as determined by the user's path. As a last resort, the current
+    python execution environment will be used (still in a subprocess).
+
+    Parameters:
+        script_source (str): The python script to be executed, as a string.
+        env_type: The type of execution environment to use (default: ExecutionEnvironmentType.AUTO_DETECT).
+        cwd (Optional[Union[str, Path]]): An optional path to use as the current directory when exeecuting the script.
+        timeout (Optional[int]): An optional timeout in seconds when executing the script. Defaults to no timeout.
+        **kwargs: Additional keyword arguments used by specific execution environments, as follows:
+            - venv environments accept a 'path' argument to specify the venv root directory.
+            - conda environments accept a 'name' argument to specify the name of the conda environment to use.
+    Returns:
+        A subprocess.CompletedProcess object containing the output of the script, if any.
+    """
+    if env_type == ExecutionEnvironmentType.AUTO_DETECT:
+        if _is_venv_active():
+            return _execute_in_venv(script_source, cwd=cwd, timeout=timeout)
+        elif _is_conda_active():
+            return _execute_in_conda_env(script_source, cwd=cwd, timeout=timeout)
+        else:
+            return _execute_with_system_path_python(
+                script_source, cwd=cwd, timeout=timeout
+            )
+    elif env_type == ExecutionEnvironmentType.VENV:
+        return _execute_in_venv(
+            script_source, kwargs.get("path"), cwd=cwd, timeout=timeout
+        )
+    elif env_type == ExecutionEnvironmentType.CONDA:
+        return _execute_in_conda_env(
+            script_source, kwargs.get("name"), cwd=cwd, timeout=timeout
+        )
+    elif env_type == ExecutionEnvironmentType.SYSTEM_PATH:
+        return _execute_with_system_path_python(script_source, cwd=cwd, timeout=timeout)
+    else:  # ExecutionEnvironmentType.CURRENT
+        return _execute_python_interpreter(
+            sys.executable, script_source, cwd=cwd, timeout=timeout
+        )
```

Comment:
Right now, the caller of `execute_script_in_sandbox` only passes in `VENV` or `CONDA`, and in the absence of both, `AUTO_DETECT` is the default.
How would `SYSTEM_PATH` (line 196) and `CURRENT` (line 198) become true? Are these supposed to come from the user as part of the `properties` of the processor?

(I do see that `_execute_with_system_path_python` is used in `AUTO_DETECT` too, in addition to line 197)

Reply:
Again, it's a general function. You use what you need, but you won't need everything right now.

Reply:
Ack